conanfile.py

```diff
@@ -10,9 +10,15 @@ class ProtobufConan(ConanFile):
     version = "2.6.1"
     url = "https://github.com/memsharded/conan-protobuf.git"
     settings = "os", "compiler", "build_type", "arch"
-    exports = "CMakeLists.txt", "*.cmake", "extract_includes.bat.in"
+    exports = "CMakeLists.txt", "*.cmake", "extract_includes.bat.in", "change_dylib_names.sh"
     options = {"static": [True, False]}
     default_options = "static=False"
+    requires = "zlib/1.2.8@lasote/stable"
+    generators = "cmake"
+    license = "https://github.com/google/protobuf/blob/v2.6.1/LICENSE"
+
+    def config(self):
```

Comment:
:+1: for renaming `static` option.

I've added this line
```python
self.options["zlib"].shared = not self.options.static
```
because same Package_IDs given for different `zlib:shared` option.

For example, when building Protobuf with different `zlib:shared` values,
```sh
conan install Protobuf/2.6.1@memsharded/testing --build Protobuf -o zlib:shared=True -o Protobuf:static=True
```
and
```sh
conan install Protobuf/2.6.1@memsharded/testing --build Protobuf -o zlib:shared=False -o Protobuf:static=True
```
the resulting IDs are both `Protobuf/2.6.1@kbinani/testing:ee5dab950b6ea94ae953b93c55343e98c2ec39df`.

This may confuse users, so my selection was fixing the combination of `zlib:shared` and `Protobuf:static`.
However, I also think it is appropriate to allow users to choose the combinations.

I think there are two solutions.

1. Adding zlib linkage option to Protobuf package itself (for example, `zlibshared=True`).
1. Changing behavior of `ConanInfo#package_id` https://github.com/conan-io/conan/blob/develop/conans/model/info.py#L187 to generate different Package_IDs for different dependent package options.